SimPEG/regularization/base.py

```diff
@@ -55,7 +55,7 @@ def __init__(
         mapping: maps.IdentityMap | None = None,
         reference_model: np.ndarray | None = None,
         units: str | None = None,
-        weights: dict | None = None,
+        weights: dict | np.ndarray | None = None,
```

Comment:
I don't think it should ever be an ndarrray. It can be right now just to accommodate the deprecation of `cell_weights`. We should probably change the init below and make it more explicit that "cell_weights" got converted to an entry of the dict.

Suggesting moving L109-110 inside the deprecation block at L98

Reply:
I see. I added that type hint because the constructor supports weights as `ndarray`. If we actually don't want this and we want users to be passing dicts to it, then we need to go through a deprecation process: if I move L109-110 inside the deprecation block in L98, then we'll break a lot of exiting code (even our own docs). In case we want to do so we would need to raise warnings for one or two releases and then deprecate it. 

I wouldn't add deprecation warnings in this PR since it's unrelated to the new feature it's being added. Would be nice to open an issue for it.

Reply:
Hmm I don't see anywhere `weights=vector` being used anywhere. Pretty joe and I did a deep cleanup of those. But sure, don't need to move the lines on this PR. Let's just not change the type hint.

Reply:
There are a few places that pass either `weights` or `cell_weights` as arrays:

https://github.com/simpeg/simpeg/blob/e3cd6ed869d5e49ab817b45f1fec037ac638a68f/examples/08-vrm/plot_inv_vrm_eq.py#L199

https://github.com/simpeg/simpeg/blob/e3cd6ed869d5e49ab817b45f1fec037ac638a68f/tests/base/test_regularization.py#L282-L286

https://github.com/simpeg/simpeg/blob/e3cd6ed869d5e49ab817b45f1fec037ac638a68f/tests/base/test_regularization.py#L320-L322

But besides that, I agree: we shouldn't add the type hint and start a deprecation process for `weights` as arrays.

Reply:
Right, first one is `cell_weights`, that's fine, but the tests could be fixed. But that's the thing, I don't think we need to deprecate weights as array since it was never intended. But you're the judge.

Reply:
Passing `cell_weights` is the same as passing `weights` + a warning, so it would break that. But understand your point. So probably we could just remove it.

I'm merging this PR then and open a new one for your suggestion of allowing `weights` only as dicts or None.